### DISCUSSION TITLE
I don't understand PossiblyReadAsUndefined when writing

### DISCUSSION THREAD

**Discussion by kriszlazar** [tylim88/FirelordJS#104] [Q&A] [answered]
Hello 👋 , thank you for this amazing library it makes working with Firestore a breeze 🙏 , 
got a question regarding `PossiblyReadAsUndefined`, when writing docs to collections: how would I be able to create a property that is "excluded" from the document's type when writing? 
ex.: given the following type:
```ts
export type Users = MetaTypeCreator<
  {
    name: string | PossiblyReadAsUndefined;
    age: number | PossiblyReadAsUndefined;
  },
  'users'
>;
```

when writing the above doc using:
```ts
const db = getFirestore();

const usersRef = getFirelord<Users>(db, 'users');

setDoc(usersRef.doc('test'), { name: 'helo' }); // shouldnt name be type name | undefined here? 🤔 (excluded)
```
I get the following error:
![image](https://github.com/tylim88/FirelordJS/assets/39099608/e0e58b7d-963f-4373-b161-c5c9a258c6e3)
or am I missing something? 🤔 
here is the stackblitz that I have used: https://stackblitz.com/edit/vitejs-vite-4jvbus?file=src%2Flib%2Fdb.ts

thank you so much 🙇‍♂️

**Comment by tylim88** [comment, marked as answer]
there is no optional field, all fields must have a default value when you create them for the first time using `setDoc`

reason behind this design decision:

> Optional is a highly requested feature because of how common it is, however because of how Firestore works: it is impossible to query a missing field. Example: it is impossible to query user that has no phone number if phone number field does not exist. Because of this, it is important to make sure every field exists. You may not need the field now, but you may need it later plus adding default value is simple, especially with such powerful typing library like Firelord. So in order to not accidentally cripple your query in the future, I will not implement this feature. Yes, set merge basically lead to the same problem, hence I discourage you from using set merge.


this is your type
```ts
export type Users = MetaTypeCreator<
  {
    name: string | null;
    age: number | null;
  },
  'users'
>;
```

this is your code
```ts
setDoc(usersRef.doc('test'),{name:'hello', age:null})
```
`PossiblyReadAsUndefined` only affect read operation https://firelordjs.com/api/possibly_read_as_undefined

**Reply by kriszlazar** [reply]
HI @tylim88 , thank you so much for the fast response, my bad totally missed this from the github readme 🙈.
